### ISSUE TITLE
Nullable property type

### ISSUE THREAD

**Issue by erictheswift** [oats-ts/oats-ts#145]
Hey! Thanks for oats!

Noticed an issue with nulls.
Given schema:
```json
{
  "type": "object",
  "required": ["id", "date"],
  "properties": {
    "id": {"type": "integer"},
    "group_id": {"type": "integer","nullable": true},
    "date": {"type": "string","nullable": true}
  }
}
```
I expect oats to generate type 
```typescript
type Expected = {id: number, group_id?: number | null, date: string | null}
```
however `nullable: true` is ignored and results in
```typescript
type Actual = {id: number, group_id?: number, date: string}
```
Workaround I've used so far: removing null properties from "required" list makes oats generate type predicates that allow null as a value, e.g. for group_id I have:
```typescript
 (input.group_id === null ||
      input.group_id === undefined ||
      typeof input.group_id === 'number') &&
```
that makes it pass the response validation.

However type definition still does not allow nulls and it hurts when
1. endpoint demand strict null as request value
2. using response data as typed test fixture — type mismatch occurs
(have to workaround with `null as undefined`)

Did not succeed with any of mentioned options:
https://stackoverflow.com/questions/48111459/how-to-define-a-property-that-can-be-string-or-null-in-openapi-swagger#48114322 — some are not parsed as valid config, some produce `string | any` instead of `string | null`

Seems there's no way to instruct oats to generate null property.

**Comment by bali182** [comment]
Fair point, I thought alternatives you mentioned (and omitting the `null` fields/values) would satisfy all use cases, but haven't considered the use-case where you explicitly want to send/receive `null`.

**Comment by bali182** [comment]
I have added a fix for this, behaviour will be:
Type guards:
- optional object properties (properties missing from the `required` array) are only matching, if they are missing or the field is set to undefined (this is unique for object properties)
- `nullable` schemas (this can happen at any level) will be accepted both as `null` and as the schema containing the `nullable` field

Types:
These will be inline with type guards, properties optional properties will be optional fields (`?:`) and nullable types will be `Type | null`

Validators:
These sadly will be more permissive for the moment, so null and undefined will be interchangeable for them. I don't want to fix this now, as there is a big rework for this on a different branch (it won't affect the public facing APIs much).

**Comment by bali182** [comment]
Released v0.0.51, which supports explicit nulls, fire a new one if you see any issues cropping up with this!

**Comment by erictheswift** [comment]
Oh thanks for fast response!
So much of
```
function hackOatsNullInconsistency<T>(_: T | null) {
  return _ as T | undefined
}
```
gone!
